public/app/common/components/occurrence-indicator/viewmodels/test/component-viewmodel.spec.coffee

```diff
@@ -0,0 +1,31 @@
+ComponentViewModel   = require '../component.viewmodel'
+PatternViewModel     = require '../pattern.viewmodel'
+{nx} = require 'nexus'
+
+describe 'ComponentViewModel', ->
+
+	view_model = null
+	data = ['this', 'is', 'test', 'data']
+
+	beforeEach ->
+		view_model = new ComponentViewModel data
+
+	describe 'constructor', ->
+
+		it 'saves passed array to `patterns`', ->
+			view_model.patterns.length.should.equal data.length
+			view_model.patterns.every (pattern) -> pattern instanceof PatternViewModel
+				.should.equal true
+
+		it 'creates an nx.Cell property with empty string value', ->
+			view_model.string.should.be.an.instanceof nx.Cell
+			view_model.string.value.should.equal ''
+
+	it 'should highlight matching patterns', ->
```

Comment:
Highlighting is a specific behavior, I'd simply go with `updates the is_matched cell for every pattern`.
`is_matched`'s name already speaks for itself and that's exactly the expected behavior